boot/bootutil/src/swap_scratch.c

```diff
@@ -174,6 +174,13 @@ boot_slots_compatible(struct boot_loader_state *state)
     j = sz1 = secondary_slot_sz = 0;
     smaller = 0;
     while (i < num_sectors_primary || j < num_sectors_secondary) {
+        // Make sure that the boot_img_sector_size() calls below don't access undefined memory.
```

Comment:
I ran into a rather nasty issue when testing where I didn't set BOOT_MAX_IMG_SECTORS large enough, so this for loop exhibited undefined behavior.  Throwing in a check against this.

Reply:
this doesn't even make sense, this is already checked on line 156 above

Reply:
Hmm, I checked in the debugger, and if I set BOOT_MAX_IMG_SECTORS to 1000, on a device that uses 1536, I see: 
![image](https://github.com/user-attachments/assets/3d450b6d-f727-4979-ab12-016316778035).

Looking into it a bit more, it seems this may be an issue due to the Mbed version of flash_area_get_sectors() never returning a `count` value greater than `MCUBOOT_MAX_IMG_SECTORS`.  That causes the existing warning (which I totally didn't notice X_X) to trigger.

Reply:
ok pushed a fix!